src/aks-preview/azext_aks_preview/decorator.py

```diff
@@ -1890,6 +1915,9 @@ def set_up_agent_pool_profiles(self, mc: ManagedCluster) -> ManagedCluster:
         agent_pool_profile.message_of_the_day = (
             self.context.get_message_of_the_day()
         )
+        agent_pool_profile.enable_custom_ca_trust = (
+            self.context.get_enable_custom_ca_trust()
+        )
```

Comment:
The test case `test_construct_mc_preview_profile` failed due to the default agentpool profile is different from it used to be. The default value of property `enable_custom_ca_trust` has changed to `False` from `None`.

Reply:
Normally, when you add an option whose default value is not `None`, you will need to change the hard-coded target managedCluster (mc) object in this test case. This test case is used to check how the CLI constructs a mc object when the user does not specify any options (which will then be sent to the ARM/RP as the request body).

Specifically in your case, in aks create, the default value of the new option `enable_custom_ca_trust` is `False`, then when the user does not explicitly set this option, your code will set the value of the property `enable_custom_ca_trust` of the `agentpool` in mc to `False` (in the above code), which is different from it used to be, that the property `enable_custom_ca_trust` of the `agentpool` is left with default value `None` when `agentpool` is initialized (see [model](https://github.com/UtheMan/azure-cli-extensions/blob/mikolaj/add-custom-ca-trust/src/aks-preview/azext_aks_preview/vendored_sdks/azure_mgmt_preview_aks/v2022_04_02_preview/models/_models_py3.py#L3416)).

Reply:
I believe these tests are now fixed :)